python/cudf/cudf/core/dataframe.py

```diff
@@ -4167,6 +4167,23 @@ def stack(self, level=-1, dropna=True):
         else:
             return result
 
+    def cov(self, **kwargs):
+        """Compute the covariance matrix of a DataFrame.
+        Parameters
+        ----------
+        **kwargs
+            Keyword arguments to be passed to cupy.cov
+        Returns
+        -------
+        cov : DataFrame
+        """
+        cov = cupy.cov(self.values, rowvar=False)
+        df = DataFrame()
+        for i, col in enumerate(self.columns):
```

Comment:
I know this is WIP, but it may be good to use `from_gpu_matrix` here in case of future optimizations of how we convert from ndarray to dataframe. Perhaps something like the following:

```python
df = DataFrame.from_gpu_matrix(
            cp.asfortranarray(result) # enforce column major
        ).set_index(...)
df.columns = ...
```

Reply:
good call - that is what `from_gpu_matrix` is for :)